server.py

```diff
@@ -49,10 +83,31 @@ def run_server():
             conn, addr = server.accept()
             msg = ''
             while True:
+                """
+                If response in msg, can use this to return Ok or Error
+                """
                 msg_recv = conn.recv(4096)
                 msg += msg_recv
                 if len(msg_recv) < 4096:
-                    conn.sendall(response_ok())
+                    # import pdb; pdb.set_trace()
+                    try:
+                        client_response = parse_request(msg)
```

Comment:
afaict, your `parse_request` method does not return a proper HTTP Response.  Nor should it.  You still need to take the return value from that function and build a good response if no error arises from calling the function.